releases.md

```diff
@@ -3,6 +3,9 @@ title: Project Calico Documentation Archives
 description: Home
 layout: docwithnav
 ---
+This page contains links to all Project Calico documentation for previous
+releases, as well as to the latest release and the nightly build of
+documentation. There is a separate page for [Release Notes](/release-notes/)
```

Comment:
Release notes are versioned, whereas this page is not.  i.e. There a release notes page in each of the versions of the docs, whereas there is just one of this page. This makes me think that the Release Notes link is likely not right.  You probably can't put a link here, and instead just have to describe that each version of the documentation has a release notes page with details of the individual releases for that version.